### ISSUE TITLE
[Demo app] : Display components in both light/dark mode

### ISSUE THREAD

**Issue by Tayebsed93** [Orange-OpenSource/ouds-flutter#223]
# Context : 
In the current demo app, each component needs to be shown in both Light and Dark themes.
Right now, this requires manually wrapping each component to force the theme, which leads to duplicated and harder-to-maintain code.

Here is an example of what should be used to force a specific theme:

```
OudsTheme(
  themeContract: OrangeTheme(),
  themeMode: ThemeMode.dark,
  onColoredSurface: false,
  child: Builder(
    builder: (context) {
      return Text(""); // Replace with the actual component
    },
  ),
),
```

# Goal

<img width="276" alt="Image" src="https://github.com/user-attachments/assets/b4e5644a-ca3e-4b86-be1e-501bdf695d59" />
 

- Ensure each component is automatically displayed in both Light and Dark mode.
- Refactor the current implementation to avoid code duplication.
- Keep the code clean, modular, and maintainable.

**Comment by B3nz01d** [comment]
Completed during the sprint 19
